go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -104,67 +107,230 @@ func UpdatePod(taskExecutionMetadata pluginsCore.TaskExecutionMetadata,
 	ApplyInterruptibleNodeAffinity(taskExecutionMetadata.IsInterruptible(), podSpec)
 }
 
-// ToK8sPodSpec constructs a pod spec from the given TaskTemplate
-func ToK8sPodSpec(ctx context.Context, tCtx pluginsCore.TaskExecutionContext) (*v1.PodSpec, error) {
-	task, err := tCtx.TaskReader().Read(ctx)
+// BuildRawPod constructs a PodSpec and ObjectMeta based on the definition passed by the TaskExecutionContext. This
+// definition does not include any configuration injected by Flyte.
+func BuildRawPod(ctx context.Context, tCtx pluginsCore.TaskExecutionContext) (*v1.PodSpec, *metav1.ObjectMeta, string, error) {
+	taskTemplate, err := tCtx.TaskReader().Read(ctx)
 	if err != nil {
 		logger.Warnf(ctx, "failed to read task information when trying to construct Pod, err: %s", err.Error())
-		return nil, err
+		return nil, nil, "", err
+	}
+
+	var podSpec *v1.PodSpec
+	objectMeta := metav1.ObjectMeta{
+		Annotations: make(map[string]string),
+		Labels:      make(map[string]string),
 	}
-	if task.GetContainer() == nil {
-		logger.Errorf(ctx, "Default Pod creation logic works for default container in the task template only.")
-		return nil, fmt.Errorf("container not specified in task template")
+	primaryContainerName := ""
+
+	if taskTemplate.GetContainer() != nil {
```

Comment:
Switch it to a `switch target := taskTemplate.GetTarget().(type)` block to avoid casting twice and maybe improve readability that it's a "oneof" ?

Reply:
Great idea, done!